cli/cmd/vuln_host.go

```diff
@@ -967,47 +903,39 @@ func hostScanPackagesVulnDetailsTable(vulns []api.HostScanPackageVulnDetails) []
 	return out
 }
 
-func hostScanPackagesVulnPackagesView(vulns []api.HostScanPackageVulnDetails) [][]string {
-	out := [][]string{}
-	for _, vuln := range vulns {
-		if vulCmdState.Fixable && vuln.HasFix() {
-			continue
-		}
-
-		added := false
-		for i := range out {
-			if out[i][1] == vuln.Severity &&
-				out[i][2] == vuln.OsPkgInfo.Pkg &&
-				out[i][3] == vuln.OsPkgInfo.PkgVer {
+func filterHostScanPackagesVulnPackages(vulns []api.HostScanPackageVulnDetails) filteredPackageTable {
+	var filteredPackages []packageTable
+	var aggregatedPackages []packageTable
 
-				if countCVEs, err := strconv.Atoi(out[i][0]); err == nil {
-					out[i][0] = fmt.Sprintf("%d", (countCVEs + 1))
-					added = true
-				}
-
-				if vuln.HasFix() {
-					if fixes, err := strconv.Atoi(out[i][4]); err == nil {
-						out[i][4] = fmt.Sprintf("%d", (fixes + 1))
-					}
-				}
-			}
+	for _, vuln := range vulns {
+		pack := packageTable{
+			cveCount:       1,
+			severity:       strings.Title(vuln.Severity),
+			packageName:    vuln.OsPkgInfo.Pkg,
+			currentVersion: vuln.OsPkgInfo.PkgVer,
 		}
 
-		if added {
+		if vulCmdState.Fixable && !vuln.HasFix() {
+			filteredPackages = aggregatePackagesWithHosts(aggregatedPackages, pack, false, false)
 			continue
 		}
 
-		fixes := "0"
-		if vuln.HasFix() {
-			fixes = "1"
-		}
+		aggregatedPackages = aggregatePackagesWithHosts(aggregatedPackages, pack, false, vuln.HasFix())
+	}
+
+	totalUnfiltered := len(filteredPackages) + len(aggregatedPackages)
+	return filteredPackageTable{packages: aggregatedPackages, totalPackages: len(aggregatedPackages), totalUnfiltered: totalUnfiltered}
```

Comment:
nit: style and no need to instantiate `totalUnfiltered` variable.
```suggestion
	return filteredPackageTable{
		packages:        aggregatedPackages,
		totalPackages:   len(aggregatedPackages),
		totalUnfiltered: len(filteredPackages) + len(aggregatedPackages),
	}
```